package.json

```diff
@@ -14,13 +14,14 @@
     "build:constants": "dotenv -- ./scripts/constants.sh",
     "build:debug": "dotenv -- ./scripts/debug.sh",
     "build:templates": "concurrently 'npm run build:constants' 'npm run build:debug'",
-    "build:typescript": "tsc -p tsconfig.build.json",
+    "build:typescript": "swc --config-file .swcrc.build -d dist *.ts **/*.ts",
     "build:citree": "cp ast/index.citree tools/citree/example/ast.citree && cd tools/citree && npm run build && npm run snap",
     "build:test-generator": "cd tools/test-generator && npm run build",
     "build:ast": "citree ast/index.citree",
     "build:test": "npm run build:test-generator && dotenv -e test.env -- test-generator ./test",
     "build:test:activate": "dotenv -e test.env -- test-generator --activate ./test",
     "check": "dotenv -e test.env -- concurrently 'npm run build:templates' 'npm run build:ast' && tsc --noEmit",
+    "check:swc": "dotenv -e test.env -- concurrently 'npm run build:templates' 'npm run build:ast' && swc -d /tmp/dist *.ts **/*.ts",
```

Comment:
I had the idea of keeping check with tsc for now, and piloting a type check that uses swc, includes test files and outputs to a temp location.

I think in terms of writing the data to a garbage location this is fine. But I'd definitely want to see if I even like the type checkout output of swc. This will let me try it out in parallel.

Also, keep in mind that node-tap still uses tsc, so I won't actually be able to fully ditch tsc until node-tap is fully removed.

Reply:
I ditchec this, because I realized swc is designed in a way where it trades correctness for speed - for real type checking, you *should* continue using tsc. In fact, if you need *type declarations*, you should continue using tsc. swc is best intended for "production builds" where all type information is both ignored and erased.